elf_ctx.c

```diff
@@ -166,12 +199,51 @@ void elf_ctx_init(struct ukarch_ctx *ctx, struct elf_prog *prog,
 		{ AT_PHNUM, prog->phdr.num },
 		{ AT_PHDR, (__uptr)prog->vabase + prog->phdr.off },
 #if CONFIG_APPELFLOADER_VDSO
+		/* TODO: This must also be pushed and copied
+		 * or mapped to information block. Move it to infoblk_auxv
+		 * ASAP.
+		 */
 		{ AT_SYSINFO_EHDR, (uintptr_t)vdso_image_addr },
 #endif /* CONFIG_APPELFLOADER_VDSO */
 		{ AT_IGNORE, 0x0 }
 	};
 	struct auxv_entry auxv_null = { AT_NULL, 0x0 };
 
+	for (i = (int)ARRAY_SIZE(infoblk_auxv) - 1; i >= 0; i--) {
```

Comment:
You are correct, even the ABI states that the order does not matter. However in linux these are indeed placed one after another and, since I cannot know the base of infoblk and only its top (like a stack), I have to to reverse iteration.

This whole PR was created after realising that nextJS computes the length of an `argv` by doing `argv[n + 1] - argv[n]` which assumes order :(.

Reply:
> order does not matter.

What is more, the ABI also states that the spacing between the argv's does not matter... and here we are :rofl: